frontend/src/components/abfragevarianten/SachbearbeitungComponent.vue

```diff
@@ -141,9 +201,71 @@ export default class AbfragevarianteSachbearbeitungFormular extends Mixins(
     return [];
   }
 
-  get isBaugenehmigungsverfahren(): boolean {
-    const abfrage: AbfrageModel = this.$store.getters["search/selectedAbfrage"];
-    return abfrage.artAbfrage === AbfrageDtoArtAbfrageEnum.Baugenehmigungsverfahren;
+  private setGroupedStammdatenList(): void {
+    const stammdaten = this.$store.getters["stammdaten/foerdermixStammdaten"];
+    this.groupedStammdaten = this.groupItemsToHeader(stammdaten);
+  }
+
+  private onIsASobonBerechnungChange(): void {
+    this.formChanged();
+    if (!this.abfragevarianteSachbearbeitung.isASobonBerechnung) {
+      this.abfragevarianteSachbearbeitung.sobonFoerdermix = undefined;
+    }
+  }
+
+  /**
+   * Gruppiert eine Liste von Fördermixstämmen nach dem Attribut'bezeichnungJahr' welche den Wert 'SoBoN 2021' und 'SoBoN 2017' entsprechen.
+   * Außerderm fügt die Methode entsprechende header-Objekte hinzu.
+   * Gedacht zum Einsatz mit v-select.
+   *
+   * @param foerdermixStaemme Eine zu gruppierende Liste von {@link FoerdermixStammModel}.
+   * @return Eine neue Liste, welche neben den Fördermixstämmen auch { header: string }-Objekte enthält.
+   */
+  groupItemsToHeader(foerdermixStaemme: FoerdermixStammModel[]): GroupedStammdaten {
```

Comment:
Handelt es sich um die selbe Logik wie im Fördermixformular? Falls ja, sollte diese Methode in eine eigene Klasse ausgelagert werden.